pkg/auth/http.go

```diff
@@ -17,12 +17,17 @@ type HttpAuthContext struct {
 func AuthMiddleware(backendRepo repository.BackendRepository, workspaceRepo repository.WorkspaceRepository) echo.MiddlewareFunc {
 	return func(next echo.HandlerFunc) echo.HandlerFunc {
 		return func(c echo.Context) error {
+			var tokenKey string
 			req := c.Request()
 			authHeader := req.Header.Get("Authorization")
-			tokenKey := strings.TrimPrefix(authHeader, "Bearer ")
+			tokenKey = strings.TrimPrefix(authHeader, "Bearer ")
 
 			if authHeader == "" || tokenKey == "" {
-				return next(c)
+				// Check for the token in the Sec-WebSocket-Protocol header
+				tokenKey = parseAuthFromWebsocketProtocolHeader(req.Header.Get("Sec-WebSocket-Protocol"))
```

Comment:
what about unauthorized websockets?

Reply:
we just extract the token, we still verify it through the normal authorize token workflow